src/link/index.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link as ReactRouterLink } from 'react-router-dom';
+
+// eslint-disable-next-line no-useless-escape
+const linkRegex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
```

Comment:
What if we passed a prop instead of checking for regex?

Reply:
I'd prefer it to be "magic" and swap the `Link` component out for an anchor, but happy to make it a prop instead.
Up to you?

Reply:
I'm unsure. It just means that we have to have a URL for href and can't use relative paths. Which maybe is a good thing? Really thinking out loud here ;)

Reply:
That's the plan. If it's a relative path, you want it to use the `Link` component, which means react-router would be used to navigate around the app.
You can't use an href to navigate in the react app.

Ideally, we'd use this component vs. `import { Link } from 'react-router-dom'` for some consistency.

Reply:
Awesome. Can you add it to link-button? :D